### PR TITLE
chore(Tooltip): move to function component

### PR DESCRIPTION
**Pull request by layershifter**
# BREAKING CHANGES

As in other components converted to hooks `TooltipContent` now passes only these props to styles: `placement`, `pointing`, `open`.

# Performance

I used an example with 100 opened `Tooltip`s to measure rendering speed. And it almost 2 times faster because of enabled caching (#2309).

```sh
yarn perf --filter *TooltipMin*
```

```tsx
const TooltipMinimalPerf = () => (
  <>
    {_.times(100, i => (
      <Tooltip key={i} content="Sample">
        <div />
      </Tooltip>
    ))}
  </>
);
```

### Before

| Example                 | min   | avg   | median | max    |
| ----------------------- | ----- | ----- | ------ | ------ |
| TooltipMinimal.perf.tsx | 74.38 | 78.47 | 78.12  | 104.65 |
| TooltipMinimal.perf.tsx | 74.92 | 78.98 | 78.19  | 98.78  |
| TooltipMinimal.perf.tsx | 74.99 | 78.18 | 77.71  | 91.64  |

### After without caching

It's also a bit faster because `Tooltip` now completely omits styles computations.

| Example                 | min   | avg   | median | max   |
| ----------------------- | ----- | ----- | ------ | ----- |
| TooltipMinimal.perf.tsx | 69.55 | 76.33 | 74.1   | 92.98 |

### After

| Example                 | min   | avg   | median | max   |
| ----------------------- | ----- | ----- | ------ | ----- |
| TooltipMinimal.perf.tsx | 33.29 | 38.66 | 36.8   | 49.7  |
| TooltipMinimal.perf.tsx | 34.18 | 37.9  | 36.77  | 49.52 |
| TooltipMinimal.perf.tsx | 33.87 | 39.07 | 36.63  | 51.7  |

### CODE REVIEW

File: packages/react/src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -68,6 +67,12 @@ export interface TooltipProps
   /** Defines whether tooltip is displayed. */
   open?: boolean
 
+  // TODO
```
Comment (mnajdova) [inline packages/react/src/components/Tooltip/Tooltip.tsx]: Ups

File: packages/react/src/components/Tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -68,6 +67,12 @@ export interface TooltipProps
...
+  onOpenChange?: (
+    e: React.MouseEvent | React.FocusEvent | React.KeyboardEvent,
+    data: TooltipProps,
+  ) => void
+
   /**
    * TODO: should this be centralized?
```
Comment (mnajdova) [inline packages/react/src/components/Tooltip/Tooltip.tsx]: Yes please

File: packages/react/src/components/Tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -98,219 +103,207 @@ export interface TooltipProps
...
+
+    initialValue: false,
+  })
+
+  const contentRef = React.useRef<HTMLElement>()
+  const pointerTargetRef = React.useRef<HTMLDivElement>()
+  const triggerRef = React.useRef<HTMLElement>()
+
+  const closeTimeoutId = React.useRef<number>()
+  // TODO: Consider `getOrGenerateIdFromShorthand()` as hook and make it SSR safe
```
Comment (mnajdova) [inline packages/react/src/components/Tooltip/Tooltip.tsx:144]: Ups

File: packages/react/src/components/Tooltip/Tooltip.tsx (hunk 2 above)
Comment (layershifter) [reply packages/react/src/components/Tooltip/Tooltip.tsx]: I don't know what this means as it was added previously 🤔 I removed it for now

File: packages/react/src/components/Tooltip/Tooltip.tsx (hunk 1 above)
Comment (layershifter) [reply packages/react/src/components/Tooltip/Tooltip.tsx]: Fixed

File: packages/react/src/components/Tooltip/Tooltip.tsx (hunk 3 above)
Comment (layershifter) [reply packages/react/src/components/Tooltip/Tooltip.tsx:144]: I want to do it later as we use this function in more components (`Dialog` at least)

File: packages/react/src/components/Tooltip/TooltipContent.tsx [hunk 4]
```diff
@@ -41,42 +53,95 @@ export interface TooltipContentProps
...
+    <ElementType
+      {...getA11Props('root', {
+        className: classes.root,
+        ...rtlTextContainer.getAttributes({ forElements: [children, content] }),
+        ...unhandledProps,
+      })}
+    >
+      {open && pointing && <div className={classes.pointer} ref={pointerRef} />}
+
+      <div {...getA11Props('root', { className: classes.content })}>
```
Comment (mnajdova) [inline packages/react/src/components/Tooltip/TooltipContent.tsx]: This should be different than root

Comment (mnajdova) [inline packages/react/src/components/Tooltip/TooltipContent.tsx]:
```suggestion
      <div {...getA11Props('content', { className: classes.content })}>
```

**Review by mnajdova — APPROVED** [review]

**Pull request merged**